File: 1720-decode-xored-array/1720-decode-xored-array.c
int* decode(int* encoded, int encodedSize, int first, int* returnSize){
    int *arr;
    int i;
    arr = (int*)malloc(sizeof(int) * (encodedSize + 1));
    arr[0] = first;
    
    for(i = 1; i < encodedSize + 1; ++i)
    {
        arr[i] = encoded[i-1] ^ arr[i-1];
    }
    
    *returnSize = encodedSize + 1;
    return arr;
}

File: 1720-decode-xored-array/1720-decode-xored-array.java
class Solution {
    public int[] decode(int[] encoded, int first) {
        int[] arr = new int[encoded.length + 1];
        arr[0] = first;
        
        for(int i = 1; i < encoded.length + 1; i++)
            arr[i] = arr[i-1] ^ encoded[i-1];
        return arr;
    }
}